From: Kan-Ru Chen <kanru@kanru.info>
Date: Mon, 23 Dec 2024 08:42:33 +0900
Subject: Send CTRL-EVENT-SIGNAL-CHANGE message to control interfaces only

The default logging level for the CTRL-EVENT-SIGNAL-CHANGE message
may be repeated many times and fill the log file or journal.

For example https://bugzilla.redhat.com/show_bug.cgi?id=2309148 and
the first few results from searching CTRL-EVENT-SIGNAL-CHANGE on the
web contain various complaints and workarounds.

Change the logging method to wpa_msg_ctrl to avoid sending frequent
messages to the syslog but still allow the message to be consumed by
control interface monitors.

Signed-off-by: Kan-Ru Chen <kanru@kanru.info>

Origin: backport, https://w1.fi/cgit/hostap/commit/?id=c330b5820eefa8e703dbce7278c2a62d9c69166a
---
 wpa_supplicant/events.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index 49917f7..7a6133c 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -6590,7 +6590,7 @@ void wpa_supplicant_event(void *ctx, enum wpa_event_type event,
 					data->eapol_rx.encrypted);
 		break;
 	case EVENT_SIGNAL_CHANGE:
-		wpa_msg(wpa_s, MSG_INFO, WPA_EVENT_SIGNAL_CHANGE
+		wpa_msg_ctrl(wpa_s, MSG_INFO, WPA_EVENT_SIGNAL_CHANGE
 			"above=%d signal=%d noise=%d txrate=%lu",
 			data->signal_change.above_threshold,
 			data->signal_change.data.signal,
